### PR TITLE
Fix no destroy on restart & ktlint & Kotlin migration

### PR DESCRIPTION
**Pull request by boaz-codota**
Ref https://codota.atlassian.net/browse/DEV-2039

### CODE REVIEW

File: src/main/java/com/tabnine/binary/BinaryProcessGateway.java [hunk 1]
```diff
@@ -36,4 +36,8 @@ public void writeRequest(String request) throws IOException {
     public boolean isDead() {
         return process == null || !process.isAlive();
     }
+
+    public void destroy() {
+        process.destroy();
```
Comment (shaiahr) [inline src/main/java/com/tabnine/binary/BinaryProcessGateway.java:41]: process can't be null?
What happens if destroy throws an exception (don't know if that can happen)?

File: src/main/java/com/tabnine/lifecycle/BinaryNotificationsLifecycle.kt [hunk 2]
```diff
@@ -11,52 +11,56 @@ import com.tabnine.binary.requests.notifications.NotificationOptions
...
-        }, BINARY_NOTIFICATION_POLLING_INTERVAL, BINARY_NOTIFICATION_POLLING_INTERVAL)
+            },
+            BINARY_NOTIFICATION_POLLING_INTERVAL, BINARY_NOTIFICATION_POLLING_INTERVAL
+        )
     }
 
-    private fun storageKey(id: String?) = "tabnine-notifications-shown-${id}"
-}
\ No newline at end of file
+    private fun storageKey(id: String?) = "tabnine-notifications-shown-$id"
```
Comment (shaiahr) [inline src/main/java/com/tabnine/lifecycle/BinaryNotificationsLifecycle.kt:65]: Hard to tell if this is the only thing that changed here. I think it's better to separate format changes into a dedicated PR.
Also, this looks like a bug for something that was never tested? Did you test it now?

File: src/test/java/com/tabnine/binary/fetch/LocalBinaryVersionsTests.kt [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    private val originalHome = System.getProperty(USER_HOME_PATH_PROPERTY)
+
+    @BeforeEach
+    fun setup() {
+        System.setProperty(USER_HOME_PATH_PROPERTY, temporaryFolder.toString())
+    }
+
+    @AfterEach
+    fun teardown() {
+        System.setProperty(USER_HOME_PATH_PROPERTY, originalHome)
```
Comment (shaiahr) [inline src/test/java/com/tabnine/binary/fetch/LocalBinaryVersionsTests.kt:46]: Shouldn't we also delete the temp folder here?

**Review by shaiahr — APPROVED** [review]
LGTM
See my questions about null and exception handling.

File: src/main/java/com/tabnine/binary/BinaryProcessGateway.java (hunk 1 above)
Comment (boaz-codota) [reply src/main/java/com/tabnine/binary/BinaryProcessGateway.java:41]: good catch

Comment (boaz-codota) [reply src/main/java/com/tabnine/binary/BinaryProcessGateway.java:41]: Nope it can't. `init` is called

File: src/main/java/com/tabnine/lifecycle/BinaryNotificationsLifecycle.kt (hunk 2 above)
Comment (boaz-codota) [reply src/main/java/com/tabnine/lifecycle/BinaryNotificationsLifecycle.kt:65]: Nothing changed in that file, I only ran `ktlint` to format the whole files

File: src/test/java/com/tabnine/binary/fetch/LocalBinaryVersionsTests.kt (hunk 3 above)
Comment (boaz-codota) [reply src/test/java/com/tabnine/binary/fetch/LocalBinaryVersionsTests.kt:46]: JUnit takes care of that https://junit.org/junit5/docs/5.4.0/api/org/junit/jupiter/api/io/TempDir.html

File: src/main/java/com/tabnine/binary/BinaryProcessGateway.java (hunk 1 above)
Comment (shaiahr) [reply src/main/java/com/tabnine/binary/BinaryProcessGateway.java:41]: We check for null in isDead so is the check redundant there or missing here?

File: src/main/java/com/tabnine/lifecycle/BinaryNotificationsLifecycle.kt (hunk 2 above)
Comment (shaiahr) [reply src/main/java/com/tabnine/lifecycle/BinaryNotificationsLifecycle.kt:65]: Looks like the curly braces around the id were deleted.

File: src/test/java/com/tabnine/binary/fetch/LocalBinaryVersionsTests.kt (hunk 3 above)
Comment (shaiahr) [reply src/test/java/com/tabnine/binary/fetch/LocalBinaryVersionsTests.kt:46]: Cool.
